M2/Macaulay2/packages/ResolutionsOfStanleyReisnerRings.m2

```diff
@@ -0,0 +1,589 @@
+ -*
+ Copyright 2020, 2020 Ashleigh Adams.
+
+ You may redistribute this file under the terms of the GNU General Public
+ License as published by the Free Software Foundation, either version 2 of
+ the License, or any later version.
+ *-
+---------------------------------------
+
+newPackage(
+    "ResolutionsOfStanleyReisnerRings",
+    Version => "0.1",
+    Date => "July 15, 2020",
+    	Authors => {{Name => "Ashleigh Adams", Email => "adams869@umn.edu", HomePage => "http://www.ashleigh-adams.com"}},
+    Headline => "Comparing resolutions of Stanley-Reisner rings and computing various systems of parameters",
+    PackageExports => {"SimplicialComplexes", "Posets", "SimplicialDecomposability"},
+    DebuggingMode => false  
+    )
+
+-------------------
+-- Exports
+------------------- 
+export {"colorfulSOP","universalSOP","colorfulPresentation","universalPresentation","barycentricSubdivision","sumBetti","equalBettiTally"}
+exportMutable("Colors")
+
+-------------------
+-- Exported Code
+-------------------
+
+-----------------------------------------
+--Barycentric Subdivision-
+-----------------------------------------
+barycentricSubdivision = method(TypicalValue => SimplicialComplex)
+barycentricSubdivision (SimplicialComplex) := D -> (
+   return orderComplex(facePoset(D),CoefficientRing=>coefficientRing(ring(D)));
+);
+
+barycentricSubdivision (List) := F -> (
+   D := simplicialComplex(F);
+   return orderComplex(facePoset(D),CoefficientRing=>coefficientRing(ring(D)));
+);
+
+---------------------------------------------------
+--Non-exported code for Colorful Version
+---------------------------------------------------
+
+-- Used in balancedColoring function, it returns the color of a vertex
+vertexColor = (a,Colors) -> (
+   for i from 0 to #Colors-1 do ( if member(a,Colors#i) then return i; );
```

Comment:
This code can be much simplified by using the function `position`.

Reply:
This was great Thanks!